tests/test_merge.py

```diff
@@ -145,3 +149,77 @@ def test_get_average_frame_count(samples: str, count: float) -> None:
 def test_collapse_stack_consider_dso(stack: str, insert_dso_name: bool, outcome_dict: Dict[str, str]) -> None:
     expected = f"program;{outcome_dict['dso_true' if insert_dso_name else 'dso_false']}"
     assert expected == _collapse_stack("program", stack, insert_dso_name)
+
+
+def parse_profiles_text(profiles_text: str) -> ProcessToProfileData:
+    parsed: ProcessToStackSampleCounters = parse_many_collapsed(profiles_text)
+    process_to_profile_data: ProcessToProfileData = dict()
+    for pid in parsed:
+        process_to_profile_data[pid] = ProfileData(parsed[pid], None, None)
+    return process_to_profile_data
+
+
+@pytest.mark.parametrize(
+    "input_dict, expected",
```

Comment:
Good test cases. One additional case that I suggest - a profile with 2 PIDs which have 2 different errors, and the merged outcome has the error for each one correctly.

Reply:
👍